check-licenses/scripts/check-licenses.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Define non-authorized licenses
+NON_AUTHORIZED_LICENSES=("GPL" "AGPL" "LGPL" "MPL")
+
+# Get all package.json dependencies
+dependencies=$(npm list --json | jq -r '.dependencies | keys[]')
+
+# Create an empty array to store the non-authorized packages
+non_authorized_packages=()
+
+# Check each dependency
+for package in $dependencies; do
+  # Get the license of the package
+  license=$(curl -s "https://registry.npmjs.org/$package" | jq -r '.license')
+
+  # Check if the license is not authorized
+  for non_auth_license in "${NON_AUTHORIZED_LICENSES[@]}"; do
+    if [[ "$license" == "$non_auth_license" ]]; then
+      non_authorized_packages+=("$package")
+      break
```

Comment:
J'ai peur que ce break arrête complètement la boucle dès qu'une dépendance qui ne va pas a été trouvée non ?

Reply:
Non, le script retourne bien une liste des dépendances qui ne rentrent pas dans la liste autorisée. Le break permet de cut et de directement passer à l'itération suivante 🤔